protocol/0053-PERP-product_builtin_perpetual_future.md

```diff
@@ -1,18 +1,48 @@
 
 ## 1. Product parameters
 
-1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
 1. `settlement_asset (Settlement Asset)`: this is used to specify the single asset that an instrument using this product settles in.
+1. `settlement_cue (Data Source: datetime)`: this data is used to indicate the earliest time at which the next `settlement_data` should be expected.
+1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
+1. `settlement_cue_auction_duration`: a time interval which specifies the duration of an auction started once settlement cue is received. The auction ends when the specified time elapses or when the settlement data is received. A value of `0s` indicates no auction.
+1. `max_settlement_gap`: a time interval which specifies the amount of time without periodic settlement after which the market will go into protective auction and remain in that mode until settlement data is received.
+1. `settlement_data_monitoring`: if set to `true` any valid `settlement_data` will be checked against the market's [price monitoring](0032-PRIM-price_monitoring.md) engine. Specifically, the incoming settlement data will be checked against market's active [price monitoring bounds](0021-MDAT-market_data_spec.md#market-data-fields), if it falls within the bounds the periodic settlement proceeds, otherwise the system behaves as if the data was never received. This implies that if the oracle serves another valid datapoint which falls within the price monitoring bounds it will be used for periodic settlement.
 
 Validation: none required as these are validated by the asset and data source frameworks.
 
+### Example specification
+
+The pseudocode below specifies a possible configuration of the built-in perpetual futures contract product. The emphasis is on modelling required properties of this product, not the exact semantics used as these will most likely differ in the implementation.
+
+```yaml
+    product: built-in perpetual futures contract
+        settlement_asset: XYZ
+        settlement_cue:
+            internal_time_oracle:
+                repeating:
+                    - every 24h from 20230201T09:30:00
+                    - every 168h from 20230203T12:00:00
+       settlement_data:
+            data_source: SignedMessage{ pubkey=0xA45e...d6 }
+            field: 'price'
+            filters: 
+                - received: >= 'settlement_cue.time'
+                - received: <= 'settlement_cue.time' + "35min"
```

Comment:
This may need to go into `0047-DSRF-data_source_filter.md` as an extra key allowed for any oracle, but let's maybe first agree here that we want it at all.